### DISCUSSION TITLE
Animated JXL Files

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7685] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

How do I convert GIF, APNG, and animated WEBP files to animated JXL files using ImageMagick?

**Comment by dlemstra** [comment]
It should be available in your version but if you experience any issues it might help to first upgrade to the most recent version and then convert the files again.

**Reply by Jay123210599** [reply]
What should the command be?

**Reply by fmw42** [reply]
```
magick image.gif -coalesce -layers optimize image.jxl
magick image.webp -coalesce -layers optimize image.jxl
magick APNG:image.png -coalesce -layers optimize image.jxl

```

From the supported formats page at https://imagemagick.org/script/formats.php#supported


> JXL | RW | JPEG XL image coding system | Requires the JPEG XL delegate library. Use -define jxl:effort=integer to set the effort and -define jxl:decoding-speed=integer to set the decoding speed. To specify a single precision floating-point format, use -define quantum:format=floating-point. Set the depth to 32 for single precision floats and 16 for half-precision.


> [JXL](https://jpeg.org/jpegxl)	RW	JPEG XL image coding system	Requires the [JPEG XL](https://gitlab.com/wg1/jpeg-xl.git) delegate library. Use -define jxl:effort=integer to set the effort and -define jxl:decoding-speed=integer to set the decoding speed. To specify a single precision floating-point format, use -define quantum:format=floating-point. Set the depth to 32 for single precision floats and 16 for half-precisio
